File: app/src/main/java/com/hqq/app/ui/activity/tab/layout/TabLayoutActivity.java
package com.hqq.app.ui.activity.tab.layout;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.view.ViewPager;

import com.hqq.core.adapter.BaseFragmentAdapter;
import com.hqq.core.ui.BaseActivity;
import com.hqq.app.R;

import butterknife.BindView;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.core_app.ui.activity.tab_layout
 * @FileName :   TabLayoutActivity
 * @Date : 2019/6/6 0006  下午 6:00
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
public class TabLayoutActivity extends BaseActivity {
    @BindView(R.id.tb_tablayout1)
    TabLayout mTbTablayout1;
    @BindView(R.id.vp_page)
    ViewPager mVpPage;
    @BindView(R.id.tb_tablayout2)
    TabLayout mTbTablayout2;
    @BindView(R.id.tb_tablayout3)
    TabLayout mTbTablayout3;
    @BindView(R.id.tb_tablayout4)
    TabLayout mTbTablayout4;

    @Override
    public int setViewId() {
        return R.layout.activity_tab_layout;
    }

    @Override
    public void initView() {
        ViewPageAdapter adapter = new ViewPageAdapter(getSupportFragmentManager());
        mVpPage.setAdapter(adapter);
        mTbTablayout1.setupWithViewPager(mVpPage);
        mTbTablayout2.setupWithViewPager(mVpPage);
        mTbTablayout3.setupWithViewPager(mVpPage);
        mTbTablayout4.setupWithViewPager(mVpPage);
    }


    class ViewPageAdapter extends BaseFragmentAdapter {

        public ViewPageAdapter(FragmentManager fm) {
            super(fm);
            mStringSparseArray.append(0, "咖啡");
            mStringSparseArray.append(1, "奶茶");
            mStringSparseArray.append(2, "菊花茶");
            mStringSparseArray.append(3, "霸王杯");
            mStringSparseArray.append(4, "冬瓜茶");
        }

        @Override
        protected Fragment newFragment(int position) {
            return new TabFragment();
        }

    }
}

File: app/src/main/java/com/easy/example/ui/tab/layout/TabLayoutActivity.kt
package com.easy.example.ui.tab.layout

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayout
import com.easy.core.adapter.BaseFragmentAdapter
import com.easy.core.ui.base.BaseActivity
import com.easy.example.R

/**
 * @Author : huangqiqiang
 * @Package : com.easy.core_app.ui.activity.tab_layout
 * @FileName :   TabLayoutActivity
 * @Date : 2019/6/6 0006  下午 6:00
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
class TabLayoutActivity : BaseActivity() {

    var mTbTablayout1:TabLayout? = null
    var mVpPage:ViewPager2? = null
    var mTbTablayout2:TabLayout? = null
    var mTbTablayout3:TabLayout? = null
    var mTbTablayout4:TabLayout? = null

    override val layoutViewId:Int
        get() = R.layout.activity_tab_layout

    override fun initConfig() {
        super.initConfig()
        iToolBar?.setStatusColor(R.color.color_main)
    }

    override fun initView() {
        mTbTablayout4 = findViewById(R.id.tb_tablayout4)
        mTbTablayout3 = findViewById(R.id.tb_tablayout3)
        mTbTablayout2 = findViewById(R.id.tb_tablayout2)
        mVpPage = findViewById(R.id.vp_page)
        mTbTablayout1 = findViewById(R.id.tb_tablayout1)


        val adapter = ViewPageAdapter(this)
        mVpPage!!.setAdapter(adapter)
        mVpPage?.offscreenPageLimit = adapter.itemCount
        adapter.setupWithViewPager(mTbTablayout1!!, mVpPage!!)
        adapter.setupWithViewPager(mTbTablayout2!!, mVpPage!!)
        adapter.setupWithViewPager(mTbTablayout3!!, mVpPage!!)
        adapter.setupWithViewPager(mTbTablayout4!!, mVpPage!!)
    }

    open internal inner class ViewPageAdapter(fragmentActivity:FragmentActivity) : BaseFragmentAdapter(fragmentActivity) {
        override fun newFragment(position:Int):Fragment {
            return TabFragment()
        }

        init {
            stringSparseArray.append(0, "咖啡")
            stringSparseArray.append(1, "奶茶")
            stringSparseArray.append(2, "菊花茶")
            stringSparseArray.append(3, "霸王杯")
            stringSparseArray.append(4, "冬瓜茶")
        }
    }
}